src/core/config/ar_data/transactional_data_config.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include "algorithms/association_rules/ar_algorithm_enums.h"
+#include "config/exceptions.h"
+#include "config/names_and_descriptions.h"
+#include "config/option.h"
+#include "config/tabular_data/input_table/option.h"
+#include "config/tabular_data/input_table_type.h"
+#include "model/table/column_index.h"
+#include "model/transaction/transactional_data.h"
+
+namespace config {
+
+struct TransactionalDataParams {
```

Comment:
I'd prefer to put this structure right to the `TransactionalData` class:
```cpp
class TransactionalData {
public:
    struct Params {
        config::InputTable input_table;
        algos::InputFormat input_format;
        model::ColumnIndex tid_column_index;
        model::ColumnIndex item_column_index;
        bool first_column_tid;
    };
```

And add `CreateFrom` method, which takes this structure as param (existing `TransactionalData::CreateFrom` you can make private to prohibit their usage).

You can also pay attention to the fact that from the point of view of the language, it is strange that the parameters object has a `CreateFrom` method.